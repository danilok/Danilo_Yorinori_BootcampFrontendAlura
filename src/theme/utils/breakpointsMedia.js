import { css } from 'styled-components';
import { breakpoints } from '../index';

export default function breakpointsMedia(cssByBreakpoints) {
  const breakpointsNames = Object.keys(cssByBreakpoints);
  return breakpointsNames
    .filter((breakpointName) => Boolean(cssByBreakpoints[breakpointName]))
    .map((breakpointName) => css`
        @media screen and (min-width : ${breakpoints[breakpointName]}px) {
          ${cssByBreakpoints[breakpointName]}
        }
    `);
}
