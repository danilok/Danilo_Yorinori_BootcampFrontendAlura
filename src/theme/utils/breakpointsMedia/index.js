import { css } from 'styled-components';
import { breakpoints } from '../../index';

export default function breakpointsMedia(cssByBreakpoints) {
  const breakpointsNames = Object.keys(cssByBreakpoints);
  return breakpointsNames
    .filter((breakpointName) => Boolean(cssByBreakpoints[breakpointName])
      && breakpoints[breakpointName] !== undefined)
    .map((breakpointName) => css`
        @media screen and (min-width: ${breakpoints[breakpointName]}px) {
          ${cssByBreakpoints[breakpointName]}
        }
    `);
}
