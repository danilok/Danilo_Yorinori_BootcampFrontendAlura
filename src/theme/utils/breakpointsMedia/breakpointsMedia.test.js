import { css } from 'styled-components';
import { breakpoints as breakpointsTheme } from '../../index';
import breakpointsMedia from '.';

describe('breakpointsMedia()', () => {
  describe('when receives no argument', () => {
    test('returns an empty array', () => {
      const argument = {};
      const breakpointsMediaResult = breakpointsMedia(argument);
      expect(breakpointsMediaResult).toEqual([]);
    });
  });
  describe('when receives an argument', () => {
    describe('with only one breakpoint', () => {
      test('returns an array with single element', () => {
        const argument = {
          xs: css`
            width: 290px;
          `,
        };
        const breakpointsMediaResult = breakpointsMedia(argument);
        expect(breakpointsMediaResult).toMatchSnapshot();
        expect(breakpointsMediaResult).toHaveLength(1);
      });
    });
    describe('with two or more breakpoints', () => {
      test('returns an array with more than one elements', () => {
        const argument = {
          xs: css`
            width: 290px;
          `,
          md: css`
            width: 288px;
          `,
        };
        const breakpointsMediaResult = breakpointsMedia(argument);
        expect(breakpointsMediaResult).toMatchSnapshot();
        expect(breakpointsMediaResult.length).toBeGreaterThan(1);
      });
    });
    describe('with all breakpoints', () => {
      test('returns an array with length equals to configured breakpoints', () => {
        const argument = {
          xs: css`
            width: 290px;
          `,
          md: css`
            width: 488px;
          `,
          sm: css`
            width: 288px;
          `,
          lg: css`
            width: 588px;
          `,
          xl: css`
            width: 788px;
          `,
        };
        const breakpointsMediaResult = breakpointsMedia(argument);
        expect(breakpointsMediaResult).toMatchSnapshot();
        expect(breakpointsMediaResult).toHaveLength(Object.keys(breakpointsTheme).length);
      });
    });
    describe('with invalid breakpoint value', () => {
      test('returns an array with only valid ones', () => {
        const argument = {
          xs: css`
            width: 290px;
          `,
          st: css`
            width: 488px;
          `,
        };
        const breakpointsMediaResult = breakpointsMedia(argument);
        expect(breakpointsMediaResult).toMatchSnapshot();
        expect(breakpointsMediaResult).toHaveLength(1);
      });
    });
  });
});
