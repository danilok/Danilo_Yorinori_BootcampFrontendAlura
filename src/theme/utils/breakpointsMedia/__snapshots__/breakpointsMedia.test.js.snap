// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpointsMedia() when receives an argument with all breakpoints returns an array with length equals to configured breakpoints 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "width:290px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "width:488px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "480",
    "px){",
    "width:288px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "992",
    "px){",
    "width:588px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "1200",
    "px){",
    "width:788px;",
    "}",
  ],
]
`;

exports[`breakpointsMedia() when receives an argument with invalid breakpoint value returns an array with only valid ones 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "width:290px;",
    "}",
  ],
]
`;

exports[`breakpointsMedia() when receives an argument with only one breakpoint returns an array with single element 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "width:290px;",
    "}",
  ],
]
`;

exports[`breakpointsMedia() when receives an argument with two or more breakpoints returns an array with more than one elements 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "width:290px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "width:288px;",
    "}",
  ],
]
`;
